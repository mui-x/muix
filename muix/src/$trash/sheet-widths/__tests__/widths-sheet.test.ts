import { atomizeRuleset } from "reactxx-styler";
import { initPlatform, Shape } from "reactxx-tests";

describe("WIDTHS", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => initPlatform(isWeb))
    let ruleset

    it("01 empty", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $widths: {
          ['0-640']: null,
          ['640-1024']: [null, {}, undefined],
          ['1024-']: {}
        },
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("02 simple", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $widths: {
          ['0-640']: {
            margin: 0,
          },
          ['640-1024']: [{
            margin: 10
          }],
          ['1024-']: [{
            margin: 20
          }, {
            margin: 10
          }]
        },
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("03 WEB only: with PSEUDO", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $widths: {
          ['0-640']: {
            $web: {
              ':hover': {
                margin: 10
              }
            }
          },
          ['640-1024']: {
            color: 'red',
            $web: {
              backgroundColor: 'gray',
              ':hover': {
                padding: 10,
                ':active': {
                  margin: 10
                }
              }
            }
          }
        },
      })
      expect(ruleset).toMatchSnapshot()
    })

    it("04 WEB only: nesting PSEUDO and $widths", () => {
      ruleset = atomizeRuleset<'Text', Shape>({
        $web: {
          ':hover': {
            $widths: {
              ['0-640']: {
                $web: {
                  ':active': {
                    margin: 10
                  }
                }
              },
            }
          }
        }
      })
      expect(ruleset).toMatchSnapshot()
    })

  }
  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))
});
