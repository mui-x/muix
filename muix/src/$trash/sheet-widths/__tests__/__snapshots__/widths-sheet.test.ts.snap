// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WIDTHS ## NATIVE ## 01 empty 1`] = `null`;

exports[`WIDTHS ## NATIVE ## 02 simple 1`] = `
Array [
  Array [
    "IF $widths: <0,640) )",
    "margin: 0 /*./$widths.0-640*/",
  ],
  Array [
    "IF $widths: <640,1024) )",
    "margin: 10 /*./$switch.640-1024[0]*/",
  ],
  Array [
    "IF $widths: <1024,MAX) )",
    "margin: 20 /*./$switch.1024-[0]*/",
  ],
  Array [
    "IF $widths: <1024,MAX) )",
    "margin: 10 /*./$switch.1024-[1]*/",
  ],
]
`;

exports[`WIDTHS ## NATIVE ## 03 WEB only: with PSEUDO 1`] = `
Array [
  Array [
    "IF $widths: <640,1024) )",
    "color: red /*./$widths.640-1024*/",
  ],
]
`;

exports[`WIDTHS ## NATIVE ## 04 WEB only: nesting PSEUDO and $widths 1`] = `null`;

exports[`WIDTHS ## WEB ## 01 empty 1`] = `null`;

exports[`WIDTHS ## WEB ## 02 simple 1`] = `
Array [
  Array [
    "@media (max-width: 639px).a { margin:0 /*./$widths.0-640*/ }",
  ],
  Array [
    "@media (min-width: 640px) and (max-width: 1023px).b { margin:10px /*./$switch.640-1024[0]*/ }",
  ],
  Array [
    "@media (min-width: 1024px).c { margin:20px /*./$switch.1024-[0]*/ }",
  ],
  Array [
    "@media (min-width: 1024px).d { margin:10px /*./$switch.1024-[1]*/ }",
  ],
]
`;

exports[`WIDTHS ## WEB ## 03 WEB only: with PSEUDO 1`] = `
Array [
  Array [
    "@media (max-width: 639px).a:hover { margin:10px /*./$widths.0-640/$web*/ }",
  ],
  Array [
    "@media (min-width: 640px) and (max-width: 1023px).b { color:red /*./$widths.640-1024*/ }",
  ],
  Array [
    "@media (min-width: 640px) and (max-width: 1023px).c { background-color:gray /*./$widths.640-1024/$web*/ }",
    "@media (min-width: 640px) and (max-width: 1023px).d:hover { padding:10px /*./$widths.640-1024/$web*/ }",
    "@media (min-width: 640px) and (max-width: 1023px).e:hover:active { margin:10px /*./$widths.640-1024/$web*/ }",
  ],
]
`;

exports[`WIDTHS ## WEB ## 04 WEB only: nesting PSEUDO and $widths 1`] = `
Array [
  Array [
    "@media (max-width: 639px).a:hover:active { margin:10px /*./$web/:hover/$widths.0-640/$web*/ }",
  ],
]
`;
